### PR TITLE
Fixed mediapipe does not return an error when it has the same name as model or DAG

### PR DESCRIPTION
**Pull request by bstrzele**

### CODE REVIEW

File: src/mediapipe_internal/mediapipegraphdefinition.cpp [hunk 1]
```diff
@@ -100,6 +100,11 @@ Status MediapipeGraphDefinition::dryInitializeTest() {
 Status MediapipeGraphDefinition::validate(ModelManager& manager) {
     SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Started validation of mediapipe: {}", getName());
     ValidationResultNotifier notifier(this->status, this->loadedNotify);
+    auto& models = manager.getModels();
+    if (std::find_if(models.begin(), models.end(), [this](auto pair) { return this->getName() == pair.first; }) != models.end()) {
+        SPDLOG_LOGGER_ERROR(modelmanager_logger, "Mediapipe graph: {} is already occupied by model or pipeline.", this->getName());
```
Comment (rasapala) [inline src/mediapipe_internal/mediapipegraphdefinition.cpp:105]:
```suggestion
        SPDLOG_LOGGER_ERROR(modelmanager_logger, "Mediapipe graph name: {} is already occupied by model or pipeline.", this->getName());
```

**Review by rasapala — APPROVED** [review]
1 minor fix.

File: src/mediapipe_internal/mediapipegraphdefinition.cpp [hunk 2]
```diff
@@ -100,6 +100,11 @@ Status MediapipeGraphDefinition::dryInitializeTest() {
 Status MediapipeGraphDefinition::validate(ModelManager& manager) {
     SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Started validation of mediapipe: {}", getName());
     ValidationResultNotifier notifier(this->status, this->loadedNotify);
+    auto& models = manager.getModels();
+    if (std::find_if(models.begin(), models.end(), [this](auto pair) { return this->getName() == pair.first; }) != models.end()) {
+        SPDLOG_LOGGER_ERROR(modelmanager_logger, "Mediapipe graph name: {} is already occupied by model or pipeline.", this->getName());
+        return StatusCode::MEDIAPIPE_GRAPH_INITIALIZATION_ERROR;
```
Comment (atobiszei) [inline src/mediapipe_internal/mediapipegraphdefinition.cpp:106]: Have separate status code for this error as we won';t be able to differentiate between MP error during graph.Initialize(), and this configuration issue.

**Review by atobiszei — CHANGES_REQUESTED** [review]
I think this is insufficient fix in few aspects. We only eliminate problem that we can't now load MP when there is already model loaded with the same name.

1) What happens when we have DAG with the same name?
2) What happens if we try to load model when there is already MP loaded with the same name?
3) What happens if we try to load DAG when there is already MP loaded with the same name?

There is also no test to confirm that this fix is working.

File: src/mediapipe_internal/mediapipegraphdefinition.cpp [hunk 3]
```diff
@@ -100,6 +100,11 @@ Status MediapipeGraphDefinition::dryInitializeTest() {
 Status MediapipeGraphDefinition::validate(ModelManager& manager) {
     SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Started validation of mediapipe: {}", getName());
     ValidationResultNotifier notifier(this->status, this->loadedNotify);
+    auto& models = manager.getModels();
+    if (std::find_if(models.begin(), models.end(), [this](auto pair) { return this->getName() == pair.first; }) != models.end() || manager.pipelineDefinitionExists(this->getName())) {
```
Comment (atobiszei) [inline src/mediapipe_internal/mediapipegraphdefinition.cpp:104]:
```suggestion
    if (manager.modelExists(this->getName()) || manager.pipelineDefinitionExists(this->getName())) {
```

Comment (atobiszei) [reply src/mediapipe_internal/mediapipegraphdefinition.cpp:104]: This way we leave thread safety as the responsibility of modelmanager. In fact we shouldn't expose getModels as public

**Review by atobiszei — APPROVED** [review]

**Pull request merged**